### PR TITLE
feat: [M3-7686] - Restricted access UX for Databases

### PR DESCRIPTION
**Pull request by zaenab-akamai**
## Description 📝
To prevent unauthorized access to specific flows and provide clearer guidance, we aim to restrict entry to users without the required permissions.

The views under /databases have been updated to disable controls for restricted users

## Changes  🔄
- "Create Database Cluster" button on the landing page (with or without entries) is disabled for users without "add_databases" access
- All buttons & inputs in the Summary, Backups, Resize, Settings tabs under Database details have been disabled for users with "read_only" access
- Added a notification for restricted access users on the database details page

## Target release date 🗓️
NA

## Preview 📷

| Before  | After   |
| ------- | ------- |
| <video src="https://github.com/user-attachments/assets/e32b78ad-9a39-4241-946a-58c62674f177" /> | <video src="https://github.com/user-attachments/assets/32d40bce-a1e9-4b07-bb44-b47a7d2a334a" /> |
|Screenshot 2024-08-20 at 10 12 41 AM | Screenshot 2024-08-20 at 10 13 07 AM |

### Verification steps
1. Login as a user with full access and add databases. Grant read_only permissions to these new resources to the restricted user 
2. Login as the restricted user and navigate to the databases page. The "Create Database Cluster" button should be disabled
3. Navigate to the details page for any database - You should see a notification. Besides that, all form inputs, buttons should be disabled.
4. Login again as the user with unrestricted access and remove all access to all the databases for the restricted user. Login as the restricted user - the empty landing page for Databases should have the create button disabled.

## As an Author I have considered 🤔

*Check all that apply*

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [ ] ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

**Review by bnussman-akamai — COMMENTED** [review]
Changes look good!

Should we also disable the "Remove" buttons on Access Controls when the user has read-only permission?

Screenshot 2024-08-26 at 10 11 04 AM

**Comment by zaenab-akamai** [comment]
> Changes look good!
> 
> Should we also disable the "Remove" buttons on Access Controls when the user has read-only permission?
> 
> Screenshot 2024-08-26 at 10 11 04 AM

Hi Banks, thanks for pointing this out! I've updated the PR

**Review by bnussman-akamai — APPROVED** [review]

**Review by jdamore-linode — APPROVED** [review]
+1 to Banks's suggestion, but otherwise looks good! Thanks @zaenab-akamai!

File: packages/manager/src/features/Databases/DatabaseDetail/DatabaseBackups/DatabaseBackups.tsx [hunk 1]
```diff
@@ -22,7 +22,12 @@ import {
 import RestoreFromBackupDialog from './RestoreFromBackupDialog';
 import { BackupTableRow } from './DatabaseBackupTableRow';
 
-export const DatabaseBackups = () => {
+interface Props {
+  disabled?: boolean;
+}
+
+export const DatabaseBackups = (props: Props) => {
+  const { disabled = false } = props;
```
[the remark this answers is not in the document]
Comment (zaenab-akamai) [reply packages/manager/src/features/Databases/DatabaseDetail/DatabaseBackups/DatabaseBackups.tsx]: I agree - looks much neater without it. I've updated the PR

**Pull request merged**